frontend/packages/data-portal/e2e/pageObjects/filters/utils.ts

```diff
@@ -1,54 +1,39 @@
-import { ApolloQueryResult } from '@apollo/client'
-import { E2E_CONFIG } from 'e2e/constants'
+import { translations } from 'e2e/constants'
+import { isString } from 'lodash-es'
 
 import {
   GetDatasetByIdQuery,
   GetDatasetsDataQuery,
   GetRunByIdQuery,
 } from 'app/__generated__/graphql'
-import { QueryParams } from 'app/constants/query'
-import { getRunById } from 'app/graphql/getRunById.server'
+import { AVAILABLE_FILES_VALUE_TO_I18N_MAP } from 'app/components/DatasetFilter/constants'
 
-import { AnnotationRowCounter, TableValidatorOptions } from './types'
+import { QueryParamObjectType, RowCounterType } from './types'
 
 export function getExpectedUrlWithQueryParams({
   url,
-  queryParamKey,
-  queryParamValue,
+  queryParamsList,
   serialize,
 }: {
   url: string
-  queryParamKey?: QueryParams
-  queryParamValue: string
+  queryParamsList?: QueryParamObjectType[]
   serialize?: (value: string) => string
 }): { expectedUrl: URL; params: URLSearchParams } {
   const expectedUrl = new URL(url)
   const params = expectedUrl.searchParams
-  if (queryParamKey) {
-    params.set(
-      queryParamKey,
-      serialize ? serialize(queryParamValue) : queryParamValue,
-    )
+  if (!queryParamsList) {
+    return { expectedUrl, params }
   }
-  return { expectedUrl, params: expectedUrl.searchParams }
-}
 
-export async function getAnnotationsTableTestData({
-  client,
-  params,
-  pageNumber = 1,
-  id = +E2E_CONFIG.runId,
-}: TableValidatorOptions & { id?: number }): Promise<
-  ApolloQueryResult<GetRunByIdQuery>['data']
-> {
-  const { data } = await getRunById({
-    client,
-    params,
-    id,
-    annotationsPage: pageNumber,
+  queryParamsList.forEach(({ queryParamKey, queryParamValue }) => {
+    if (queryParamKey) {
```

Comment:
The tests pass undefined as the key, so `queryParamsList` is defined, but `queryParamKey` is not. Maybe we should rethink this, but I'm not sure.  There could be a list with multiple params and we would want to be able to remove one of them but leave the other.  We would still want something like this to work:
```
queryParamsList: [{
    queryParamKey: 'something',
    queryParamValue: 'someValue',
},
{
    queryParamKey: undefined,
    queryParamValue: '',
}]
```

Reply:
we decided to continue allowing this to be undefined.  It makes the test easier to understand because it explicitly shows that the param is no longer part of the query.